docs/development/extension-registry-cache.md

```diff
@@ -17,27 +17,23 @@ This section outlines how the extension enablement/reconciliation works, e.g., t
-   1. The webhook appends the [configure-containerd-registries.sh](../../pkg/webhook/cache/scripts/configure-containerd-registries.sh) script to the OperatingSystemConfig files. The script accepts registries in the format `<upstream_host>,<registry_cache_endpoint>,<upstream_url>` separated by a space. For each given registry, the script waits until the given registry is available (a request to the `<registry_cache_endpoint>` succeeds). Then it creates a `hosts.toml` file for the given `<upstream_host>`. In short, the `hosts.toml` file instructs containerd to first try to pull images for the given `<upstream_host>` from the configured `<registry_cache_endpoint>`. For more information about containerd registry configuration, see the [containerd documentation](https://github.com/containerd/containerd/blob/main/docs/hosts.md). The motivation to introduce the `configure-containerd-registries.sh` script is that we need to create the `hosts.toml` file when the corresponding registry is available. For more details, see [gardener/gardener-extension-registry-cache#68](https://github.com/gardener/gardener-extension-registry-cache/pull/68).
-   1. The webhook appends the `configure-containerd-registries.service` unit to the OperatingSystemConfig units. The webhook fetches the Extension resource, and then it configures the unit to invoke the `configure-containerd-registries.sh` script with the registries from the Extension status.
+   1. The webhook appends or updates `RegistryConfig` entries in the [OperatingSystemConfig CRI](https://github.com/gardener/gardener/blob/master/docs/extensions/operatingsystemconfig.md#cri-support) configuration that corresponds to configured registry caches in the Shoot. The `RegistryConfig` readiness probe is enabled so that [gardener-node-agent](https://github.com/gardener/gardener/blob/master/docs/concepts/node-agent.md) creates a `hosts.toml` containerd registry configuration file when all `RegistryConfig` hosts are reachable.
 
 ### Extension Disablement
 
 This section outlines how the extension disablement works, i.e., the extension has to be removed from the Shoot spec.
 
 1. As part of the Shoot reconciliation flow, the gardenlet destroys the [Extension](https://github.com/gardener/gardener/blob/master/docs/extensions/extension.md) resource because it is no longer needed.
-   1. If the Extension resource contains registries in its status, the registry-cache extension deploys the [`registry-cleaner` DaemonSet](../../pkg/component/registryconfigurationcleaner/registry_configuration_cleaner.go) to the Shoot cluster to clean up the existing registry configuration.
    1. The extension deletes the ManagedResource containing the registry cache resources.
```

Comment:
You could add details that GNA cleans up `hosts.toml` files and registry specific dir under `/etc/containerd/certs.d` for the registries configured. You could also add details that the extension is disabled => it no longer mutates the OSC => gardenlet updates the OSC, extension does not mutate it and this is how the registry configs from the OSC are removed. Then GNA kicks in, detects that the registries are removed and cleans up the hosts.toml files and certs.d dirs.